src/coreComponents/mesh/MeshManager.cpp

```diff
@@ -61,21 +62,32 @@ void MeshManager::generateMeshes( DomainPartition & domain )
   forSubGroups< MeshGeneratorBase >( [&]( MeshGeneratorBase & meshGen )
   {
     MeshBody & meshBody = domain.getMeshBodies().registerGroup< MeshBody >( meshGen.getName() );
-    meshGen.generateMesh(meshBody); meshBody.createMeshLevel( 0 );
+    meshBody.createMeshLevel( 0 );
+    MeshLevel & meshLevel = meshBody.getBaseDiscretization();
     CellBlockManager & cellBlockManager = meshBody.registerGroup< CellBlockManager >( keys::cellManager );
 
+    meshGen.generateMesh( cellBlockManager );
```

Comment:
I think that here we need to get something like
```suggestion
    CellBlockManagerABC const * cellBlockManager = meshGen.generateMesh();
```
such that you do no see the implementation `CellBlockManager`.

Reply:
Can't do so because Group(name, parent) needs parent for the constructor which is here meshBody, either we have CellBlockManager or meshBody as parameter of generateMesh() or I missed a thing.